src/open_inwoner/configurations/bootstrap/kic.py

```diff
@@ -0,0 +1,176 @@
+from django.conf import settings
+
+import requests
+from django_setup_configuration.configuration import BaseConfigurationStep
+from django_setup_configuration.exceptions import SelfTestFailed
+from zgw_consumers.constants import APITypes, AuthTypes
+from zgw_consumers.models import Service
+
+from open_inwoner.openklant.clients import build_client
+from open_inwoner.openklant.models import OpenKlantConfig
+from open_inwoner.utils.api import ClientError
+
+
+class KlantenAPIConfigurationStep(BaseConfigurationStep):
+    """
+    Configure the required Service to establish a connection with the Klanten API
+    """
+
+    verbose_name = "Klanten API configuration"
+    required_settings = [
+        "KLANTEN_API_ROOT",
+        "KLANTEN_API_CLIENT_ID",
+        "KLANTEN_API_SECRET",
+    ]
+    enable_setting = "KIC_API_CONFIG_ENABLE"
+
+    def is_configured(self) -> bool:
+        return bool(Service.objects.filter(api_root=settings.KLANTEN_API_ROOT))
+
+    def configure(self):
+        organization = settings.OIP_ORGANIZATION or settings.ENVIRONMENT
+        org_label = f"Open Inwoner {organization}".strip()
+
+        # 1. Create Klanten API service
+        klanten_service, created = Service.objects.update_or_create(
+            api_root=settings.KLANTEN_API_ROOT,
+            defaults={
+                "label": "Klanten API",
+                "api_type": APITypes.kc,
+                "oas": settings.KLANTEN_API_ROOT,
+                "auth_type": AuthTypes.zgw,
+                "client_id": settings.KLANTEN_API_CLIENT_ID,
+                "secret": settings.KLANTEN_API_SECRET,
+                "user_id": settings.KLANTEN_API_CLIENT_ID,
+                "user_representation": org_label,
+            },
+        )
+        if not created:
+            klanten_service.oas = settings.KLANTEN_API_ROOT
+            klanten_service.client_id = settings.KLANTEN_API_CLIENT_ID
+            klanten_service.secret = settings.KLANTEN_API_SECRET
+            klanten_service.user_id = settings.KLANTEN_API_CLIENT_ID
+            klanten_service.user_representation = org_label
+            klanten_service.save()
+
+    def test_configuration(self):
+        """
+        actual testing is done in final step
+        """
+
+
+class ContactmomentenAPIConfigurationStep(BaseConfigurationStep):
+    """
+    Configure the required Service to establish a connection with the Contactmomenten API
+    """
+
+    verbose_name = "Contactmomenten API configuration"
+    required_settings = [
+        "CONTACTMOMENTEN_API_ROOT",
+        "CONTACTMOMENTEN_API_CLIENT_ID",
+        "CONTACTMOMENTEN_API_SECRET",
+    ]
+    enable_setting = "KIC_API_CONFIG_ENABLE"
+
+    def is_configured(self) -> bool:
+        return bool(Service.objects.filter(api_root=settings.CONTACTMOMENTEN_API_ROOT))
+
+    def configure(self):
+        organization = settings.OIP_ORGANIZATION or settings.ENVIRONMENT
+        org_label = f"Open Inwoner {organization}".strip()
```

Comment:
I'm not sure where this is used, but do we want to hardcode "Open Inwoner" into the organization label? This doesn't seem to be configurable, so might not be desirable.

Reply:
It's the human readable user identification shown in the audit trail, so when Open Inwoner makes a request using an application to i.e. Zaken API, that request is logged by Zaken API and this label is associated with the "user" (application) than made the request.

I could make the whole label configurable, but I'm not sure what else a municipality would want to show here besides the name of the application (Open Inwoner) and their organization name.